### PR TITLE
Add luacheck to check builtin

### PR DESCRIPTION
**Pull request by rubenwardy**

### CODE REVIEW

**Comment by sofar** [comment]
Great, but, I would add the .luacheckrc in it's final form in a single commit, and then additional commits fixing ranges of issues (e.g. trivial ones like whitespace, globals in another, rest in a separate one).

I don't think it's a problem really, but in this stack of patches it's likely you'll make regression testing difficult due to the large amount of noise that it generates.

OTOH it's great to see these issues fixed.

File: builtin/game/chatcommands.lua [hunk 1]
```diff
@@ -367,35 +369,35 @@ core.register_chatcommand("teleport", {
...
-		local teleportee = nil
 		local p = {}
 		p.x, p.y, p.z = string.match(param, "^([%d.-]+)[, ] *([%d.-]+)[, ] *([%d.-]+)$")
-		p.x = tonumber(p.x)
-		p.y = tonumber(p.y)
-		p.z = tonumber(p.z)
 		if p.x and p.y and p.z then
+			p.x = tonumber(p.x)
+			p.y = tonumber(p.y)
+			p.z = tonumber(p.z)
```
Comment (SmallJoker) [inline builtin/game/chatcommands.lua]: You need to adjust `string.match` to be more accurate to guarantee this function call does not return `nil`. Simplest solution would again be to move it above the condition.

File: .luacheckrc [hunk 2]
```diff
@@ -0,0 +1,76 @@
+unused_args = false
+allow_defined_top = true
+
+ignore = {
+	"131", -- Unused global variable
+	"431", -- Shadowing an upvalue
+	"432", -- Shadowing an upvalue argument
+}
+
+unused_args = false
```
Comment (pauloue) [inline .luacheckrc]: This line is duplicated

**Comment by rubenwardy** [comment]
> Great, but, I would add the .luacheckrc in it's final form in a single commit, and then additional commits fixing ranges of issues (e.g. trivial ones like whitespace, globals in another, rest in a separate one).

So, my technique here was to create an initial luacheck file in one commit, then go module by module in other commits to fix issues (which also requires editing luacheck)

**Pull request merged**